src/mappings/graphPayments.ts

```diff
@@ -0,0 +1,25 @@
+import { addresses } from "../../config/addresses"
+import { GraphPaymentCollected } from "../types/GraphPayments/GraphPayments"
+import { createOrLoadEpoch, createOrLoadGraphNetwork, createOrLoadPaymentSource } from "./helpers/helpers"
+
+export function handleGraphPaymentCollected(event: GraphPaymentCollected): void {
+    let graphNetwork = createOrLoadGraphNetwork(event.block.number, event.address)
+
+    // Update epoch
+    let epoch = createOrLoadEpoch(
+        addresses.isL1 ? event.block.number : graphNetwork.currentL1BlockNumber!,
+    )
+    epoch.taxedQueryFees = epoch.taxedQueryFees.plus(event.params.tokensProtocol)
+    epoch.save()
+
+    // update graph network
+    graphNetwork.totalTaxedQueryFees = graphNetwork.totalTaxedQueryFees.plus(event.params.tokensProtocol)
+    graphNetwork.save()
+
+    // Replicate for payment source specific data
+    let paymentSource = createOrLoadPaymentSource(event.params.payer)
```

Comment:
Worth noting here that `event.params.payer` is the caller of `GraphPayments.collect()`. In most cases this will be the `PaymentsEscrow` contract. Not to be confused with the "payer" entity as the one that originally provided the funds to the `PaymentsEscrow`. 

I don't think code here is incorrect, just noting that it could be misleading.

Reply:
Added a small comment for this, might revisit this later as I don't know exactly how to make it work like it used to in V1. cc: @tmigone